test/suite/test_compact04.py

```diff
@@ -50,40 +49,62 @@ class test_compact04(wttest.WiredTigerTestCase):
-            for i in range(self.delete_range_len):
-                c.set_key(start + i)
-                c.remove()
-        c.close()
+        for iteration in range(0, 10):
+            table_uri = self.table_uri_prefix + str(iteration)
 
-        # Compact!
-        self.session.compact(self.table_uri, None)
-        self.ignoreStdoutPatternIfExists('WT_VERB_COMPACT')
-        self.ignoreStderrPatternIfExists('WT_VERB_COMPACT')
+            # Create the table and populate it with a lot of data
+            self.session.create(table_uri, self.create_params)
+            c = self.session.open_cursor(table_uri, None)
+            for k in range(self.table_numkv):
+                c[k] = ('%07d' % k) + '_' + 'abcd' * ((self.value_size // 4) - 2)
+            c.close()
+            self.session.checkpoint()
 
-        # Verify the compact progress stats.
-        c_stat = self.session.open_cursor('statistics:' + self.table_uri, None, 'statistics=(all)')
-        pages_rewritten = c_stat[stat.dsrc.btree_compact_pages_rewritten][2]
-        pages_rewritten_expected = c_stat[stat.dsrc.btree_compact_pages_rewritten_expected][2]
-        c_stat.close()
+            # Now let's delete a lot of data ranges. Create enough space so that compact runs in more
+            # than one iteration.
+            c = self.session.open_cursor(table_uri, None)
+            for r in range(self.delete_ranges_count):
+                start = r * self.table_numkv // self.delete_ranges_count
+                for i in range(self.delete_range_len):
+                    c.set_key(start + i)
+                    c.remove()
+            c.close()
 
-        self.assertGreater(pages_rewritten, 0)
-        self.assertGreater(pages_rewritten_expected, 0)
+            # Compact!
+            self.session.compact(table_uri, None)
+            self.ignoreStdoutPatternIfExists('WT_VERB_COMPACT')
+            self.ignoreStderrPatternIfExists('WT_VERB_COMPACT')
 
-        d = abs(pages_rewritten - pages_rewritten_expected) / pages_rewritten
-        self.assertLess(d, 0.15)
+            # Verify the compact progress stats.
+            c_stat = self.session.open_cursor('statistics:' + table_uri, None, 'statistics=(all)')
+            pages_rewritten = c_stat[stat.dsrc.btree_compact_pages_rewritten][2]
+            pages_rewritten_expected = c_stat[stat.dsrc.btree_compact_pages_rewritten_expected][2]
+            c_stat.close()
+
+            self.assertGreater(pages_rewritten, 0)
+            self.assertGreater(pages_rewritten_expected, 0)
+
+            d = abs(pages_rewritten - pages_rewritten_expected) / pages_rewritten
+
+            # Check whether we succeeded. Terminate the test early if things are going well.
+            message = 'Compacting %s: Prediction error: %0.2f%%' % (table_uri, d * 100)
+            if d < 0.15:
+                self.pr(message)
+                num_successes += 1
+                if num_successes >= 4 and num_failures == 0:
```

Comment:
Would one success be enough? i.e.: Could we only retry if there is a failure, but if the first run leads to a failure try again and pass if it succeeds?